### PR TITLE
Make copying optional when sending data

### PR DESCRIPTION
**Pull request by erikprice**
For context, see #424 and #426.

### CODE REVIEW

File: SocketRocket/SRWebSocket.m [hunk 1]
```diff
@@ -644,16 +644,22 @@ - (BOOL)sendString:(NSString *)string error:(NSError **)error
...
+        NSString *message = @"Invalid State: Cannot call `sendDataNoCopy:error:` until connection is open.";
         if (error) {
             *error = SRErrorWithCodeDescription(2134, message);
         }
         SRDebugLog(message);
         return NO;
     }
 
-    data = [data copy];
+    NSAssert(self.readyState != SR_CONNECTING, @"Invalid State: Cannot call `sendDataNoCopy:error:` until connection is open");
```
Comment (nlutsenko) [inline SocketRocket/SRWebSocket.m]: Probably this is no longer needed, since we have converted to an error pointer model.

**Comment by nlutsenko** [comment]
One nit, otherwise looks great.
Will merge in after tests pass and a few more sanity checks (later today).

File: SocketRocket/SRWebSocket.m (hunk 1 above)
Comment (erikprice) [inline SocketRocket/SRWebSocket.m]: Ok, removed.

File: SocketRocket/SRWebSocket.m [hunk 2]
```diff
@@ -786,7 +791,10 @@ - (void)closeConnection;
 
 - (void)_handleFrameWithData:(NSData *)frameData opCode:(NSInteger)opcode;
 {
-    frameData = [frameData copy];
+    //frameData will be copied before passing to handlers
+    //otherwise there can be misbehaviours when value at the pointer is changed
+    data = [frameData copy];
```
Comment (nlutsenko) [inline SocketRocket/SRWebSocket.m]: `frameData = [frameData copy];`
Otherwise - it doesn't quite build.

**Comment by nlutsenko** [comment]
One more typo and we are all good to merge!

File: SocketRocket/SRWebSocket.m (hunk 2 above)
Comment (erikprice) [inline SocketRocket/SRWebSocket.m]: Whoops!

**Pull request merged**